src/content-script/utils/player.ts

```diff
@@ -43,47 +45,81 @@ function findVideoPlayer(): HTMLVideoElement | null {
   return player;
 }
 
+function getPlayerInfo(player: HTMLVideoElement): VideoEvent {
+  // collect player info
+  const { paused, currentTime, duration, playbackRate } = player;
+
+  // collect url
+  let videoId = parseUrl(document.location.href).query.v;
+  if (!videoId) videoId = 'null';
+  if (Array.isArray(videoId)) [videoId] = videoId;
```

Comment:
I'm curious, what function does encapsulating the `videoId` in bracket do?

Reply:
I really wish I didn't have to write that line like that, but the linter was complaining :)

There's two things going on here
1. It's necessary in the first place because `parseUrl` returns `null` or `string` or `string[]`, but `videoId` needs to be of type `string`. Line 54 handles the null case, 55 handles the array case.
2. That weird bracket thing is because the linter doesn't like it when I did `if (Array.isArray(videoId)) videoId = videoId[0]`. It makes you use [array destructuring](https://www.freecodecamp.org/news/array-and-object-destructuring-in-javascript/). You may have already run into this but in case you haven't it's just setting `videoId` to the first element of `videoId`.

Reply:
Thanks for the clarification!